### PR TITLE
flake8: DeprecationWarning: SelectableGroups dict interface is deprec…

### PR DESCRIPTION
**Pull request by fujitatomoya**
…ated.

address https://github.com/ros2/ros2cli/issues/867

### CODE REVIEW

**Review by fujitatomoya — COMMENTED** [review]
since we cannot control `flake8` implementation, this suppresses the warning in our test code. this makes all package test green.

```bash
root@tomoyafujita:~/ros2_ws/colcon_ws# colcon test --event-handlers console_direct+ --packages-select ros2component
Starting >>> ros2component
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
cachedir: /root/ros2_ws/colcon_ws/build/ros2component/.pytest_cache
rootdir: /root/ros2_ws/colcon_ws/src/ros2/ros2cli, configfile: pytest.ini
plugins: ament-xmllint-0.16.0, ament-copyright-0.16.0, ament-pep257-0.16.0, ament-flake8-0.16.0, ament-lint-0.16.0, rerunfailures-10.2, cov-3.0.0, mock-3.6.1, repeat-0.9.1, colcon-core-0.15.0, timeout-2.1.0
timeout: 900.0s
timeout method: thread
timeout func_only: False
collecting ...                              
collected 6 items                                                              

test/test_api.py ..                                                      [ 33%]
test/test_copyright.py .                                                 [ 50%]
test/test_flake8.py .                                                    [ 66%]
test/test_pep257.py .                                                    [ 83%]
test/test_xmllint.py .                                                   [100%]

-- generated xml file: /root/ros2_ws/colcon_ws/build/ros2component/pytest.xml --
============================== 6 passed in 1.54s ===============================
Finished <<< ros2component [2.24s]          

Summary: 1 package finished [3.12s]
```

**Comment by fujitatomoya** [comment]
CC: @clalancette @iuhilnehc-ynos

**Comment by iuhilnehc-ynos** [comment]
How about adding the `pytest.ini` to filter the warning in each projects?

```
[pytest]
filterwarnings =
    error
    ignore:SelectableGroups:DeprecationWarning
```

NOTE: I tested the `ros2component` and it worked as expected.

File: ros2pkg/ros2pkg/resource/ament_python/test_flake8.py.em [hunk 1]
```diff
@@ -12,13 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import warnings
```
Comment (iuhilnehc-ynos) [inline ros2pkg/ros2pkg/resource/ament_python/test_flake8.py.em:15]: Maybe it's users' resposibility to create `pytest.ini` for their packages if they want to filter the warning.

**Review by iuhilnehc-ynos — APPROVED** [review]
I have no idea if the `python3-flake8` in the jammy will remove the warning or not.
(Maybe it's the deprecation cycle for the jammy, and it will remove in the latter Ubuntu distribution.)

This PR will fix the warning issue, so I'd like to aprrove it.

**Comment by clalancette** [comment]
> I have no idea if the `python3-flake8` in the jammy will remove the warning or not.

I suspect not.  It's been there for 2 years, so I don't think that it is going to be fixed in Ubuntu until 24.04.

**Comment by clalancette** [comment]
> How about updating the `pytest.ini` to filter the warning?

I think we should try doing this to see if it works.  That will just make it easier to remove the filter for Ubuntu 24.04.

File: ros2pkg/ros2pkg/resource/ament_python/test_flake8.py.em (hunk 1 above)
Comment (fujitatomoya) [reply ros2pkg/ros2pkg/resource/ament_python/test_flake8.py.em]: Nice 👍 this is much cleaner and better maintenance. i just stole your idea 🚀 (added you as co-author on this) see https://github.com/ros2/ros2cli/pull/868/commits/e545a5f94bf03e31bda68553833cfc2ddf146c0b

**Comment by fujitatomoya** [comment]
@iuhilnehc-ynos @clalancette `pytest.ini` just does the magic. requesting approval.

File: pytest.ini [hunk 2]
```diff
@@ -2,3 +2,4 @@
 junit_family=xunit2
 timeout=900
 timeout_method=thread
+filterwarnings=ignore:SelectableGroups:DeprecationWarning
```
[the remark this answers is not in the document]
Comment (fujitatomoya) [reply pytest.ini:6]: :+1: added with https://github.com/ros2/ros2cli/pull/868/commits/ec32e65f34d5afa1cb3162fd357a1beaa5fc08db

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19718/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14227/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20446/)

**Review by clalancette — APPROVED** [review]

**Pull request merged**